tesseract/tesseract_collision/include/tesseract_collision/core/common.h

```diff
@@ -119,24 +119,21 @@ inline ContactResult* processResult(ContactTestData& cdata,
   if (cdata.req.is_valid && !cdata.req.is_valid(contact))
     return nullptr;
 
+  if (!(contact.distance < cdata.collision_margin_data.getPairCollisionMarginData(key.first, key.second)))
+    return nullptr;
+
```

Comment:
Should this go inside the if (!found) to avoid the lookup?

Reply:
I don't believe so because every if branch seems to be performing this check or am I missing one?

Reply:
With the current layout I think you would need 2. One inside `if(!found)` and one afterward. The thought was that if it is a FIRST type and found = true, then we don't want to call getPairCollisionMarginData because it doesn't matter.